src/octoprint/util/comm.py

```diff
@@ -3122,9 +3157,8 @@ def _send_loop(self):
 		sending (through received ``ok`` responses from the printer's firmware.
 		"""
 
-		self._clear_to_send.wait()
-
-		while self._send_queue_active:
+		def _send_one_from_queue():
```

Comment:
I think this merits an extraction into a proper member function of the class.

It also lacks documentation of what the return values `True` and `False` signify. This needs to be changed, otherwise maintenance further down will become a nightmare. E.g.

``` python
def _send_one_from_queue(self):
    """
    Attempts to send one item from the send queue.

    Returns:
        (bool) True if no line was sent, False otherwise.
    """

    # ...
```

Side note: If the above logic is correct, it feels a bit backwards. I'd expect a method called `send_one_from_queue` to return `True` if it actually *does* send something, and `False` if not, so if I understand this correctly, this needs to be reversed.